Fix flaky test: `test_mapped_task_upstream_removed_with_all_failed_trigger_rules`

### Body

```
  _ TestTriggerRuleDep.test_mapped_task_upstream_removed_with_all_failed_trigger_rules[True] _
  airflow-core/tests/unit/ti_deps/deps/test_trigger_rule_dep.py:1136: in test_mapped_task_upstream_removed_with_all_failed_trigger_rules
      _test_trigger_rule(ti=ti, session=session, flag_upstream_failed=flag_upstream_failed)
  airflow-core/tests/unit/ti_deps/deps/test_trigger_rule_dep.py:1647: in _test_trigger_rule
      assert ti.state == expected_ti_state
  E   AssertionError: assert equals failed
  E     <TaskInstanceState.UPSTREAM_FAI  None                            
  E     LED: 'upstream_failed'>

```

Have seen this to be occasionally flaky. Example: https://github.com/apache/airflow/actions/runs/14963291599


### Committer

- [x] I acknowledge that I am a maintainer/committer of the Apache Airflow project.

--- a/airflow-core/tests/unit/ti_deps/deps/test_trigger_rule_dep.py
+++ b/airflow-core/tests/unit/ti_deps/deps/test_trigger_rule_dep.py
@@ -1104,6 +1104,7 @@ class TestTriggerRuleDep:
             expected_ti_state=expected_ti_state,
         )
 
+    @pytest.mark.flaky(reruns=3, reruns_delay=1)
     @pytest.mark.parametrize("flag_upstream_failed", [True, False])
     def test_mapped_task_upstream_removed_with_all_failed_trigger_rules(
         self,

